modo/api.py

```diff
@@ -370,3 +377,28 @@ def enrich_metadata(self):
                     self.update_element(inst_names[ele.name], ele)
                 else:
                     continue
+
+    def stream_cram(
+        self, cram_path: str, region: str = None, output: str = None
+    ):
+        """Slices and streams the requested CRAM file, both local and remote,
+        and either outputs a data stream or writes data to local file"""
+
+        # check requested CRAM exists in MODO
+        if Path(cram_path) not in self.list_files():
+            raise ValueError(f"{cram_path} not found in {self.path}.")
+
+        if self.s3_endpoint:
+            # http://domain/s3 + bucket/modo/file.cram --> http://domain/htsget/reads/modo/file.cram
+            url = (
+                self.htsget_endpoint
+                + "/reads/"
+                + str(Path(*Path(cram_path).parts[1:]))
+            )
+            # str(Path(*Path(cram_path).parts[1:])) same as path.split("/", maxsplit=1)[1] but cross-platform
+            slice_remote_cram(url, region, output)
+        else:
+            # assuming user did not change directory, filepath should be the
+            # relative path to the file.
+            iter = slice_cram(cram_path, region)
```

Comment:
__Suggestion__: Similar to the above point, we are not consistent between local and remote behaviour here, but only allow writing the output to an output file for remote streaming. What about adding `output` to local slicing as well?

Reply:
Added 'output_filename' argument (optional) to slice_cram(), to allow the user to copy the slice into a local file (see commit - [bcc16f3a982e8461e43f439195e34db9dc2988a0](https://github.com/sdsc-ordes/modo-api/commit/bcc16f3a982e8461e43f439195e34db9dc2988a0))

Reply:
Added argument and functionality to save_cram() MODO method (see commit - https://github.com/sdsc-ordes/modo-api/pull/31/commits/0b0ed93e2029420d128e1c81b0ce374ec7b128f3)